gpytorch/kernels/linear_kernel.py

```diff
@@ -85,12 +85,8 @@ def _set_variance(self, value: Union[float, Tensor]):
             value = torch.as_tensor(value).to(self.raw_variance)
         self.initialize(raw_variance=self.raw_variance_constraint.inverse_transform(value))
 
-    def forward(
-        self, x1: Tensor, x2: Tensor, diag: Optional[bool] = False, last_dim_is_batch: Optional[bool] = False, **params
-    ) -> LinearOperator:
+    def forward(self, x1: Tensor, x2: Tensor, diag: Optional[bool] = False, **params) -> LinearOperator:
```

Comment:
```suggestion
    def forward(self, x1: Tensor, x2: Tensor, diag: bool = False, **params) -> LinearOperator:
```
Confusingly the `Optional` type hint should only be used if `None` is a valid value, not if the argument is optional. See also the [documentation on `typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional:

> Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional.